pyanaconda/modules/network/network.py

```diff
@@ -637,18 +637,26 @@ def get_dracut_arguments(self, iface, target_ip, hostname, ibft):
             log.error("Get dracut arguments for %s: device not found", iface)
             return dracut_args
 
-        connections = self.nm_client.get_connections()
-
         target_connections = []
         if ibft:
-            target_connections = [con for con in connections if is_ibft_connection(con)]
+            target_connections = [con for con in self.nm_client.get_connections()
+                                  if is_ibft_connection(con)]
         else:
-            for cfg in self._device_configurations.get_for_device(iface):
-                uuid = cfg.connection_uuid
-                if uuid:
-                    connection = self.nm_client.get_connection_by_uuid(uuid)
-                    if connection:
-                        target_connections.append(connection)
+            if self._device_configurations:
```

Comment:
Would this work as `elif` instead of nesting?

Reply:
Hm, I wanted the ibft and non-ibft branch to be apparent so I'd keep two levels here. I mean we are not dealing with 3 cases here, but with 2 cases of which one is a bit complicated.